include/fastrtps/rtps/history/ReaderHistory.h

```diff
@@ -92,6 +92,8 @@ class ReaderHistory : public History
 
     RTPS_DllAPI bool get_min_change_from(CacheChange_t** min_change, const GUID_t& writerGuid);
 
+    void set_reader(RTPSReader* reader, std::recursive_timed_mutex* mp_mutex);
```

Comment:
I believe it should be better to change the RTPSReader mock, adding a constructor accepting an ReaderHistory pointer (default nullptr). In case the pointer is not nullptr, set itself in the ReaderHistory object.